mock_drivers/cpp_mock_drivers/test/BagParserTest.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2019-2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <gmock/gmock.h>
+#include "cpp_mock_drivers/BagParser.h"
+#include <carma_simulation_msgs/BagData.h>
+
+
+namespace mock_drivers{
+
+    TEST(BagParser, Constructor){
+        BagParser d1("");
+        BagParser d2("", false);
+        BagParser d3("", true);
+
+        ASSERT_EQ(d1.getMockDriverNode().isDummy(), false);
+        ASSERT_EQ(d2.getMockDriverNode().isDummy(), false);
+        ASSERT_EQ(d3.getMockDriverNode().isDummy(), true);
+    }
+
+    TEST(BagParser, publishCallback){
+        BagParser d("/workspaces/carma_ws/carma/src/carma-platform/mock_drivers/cpp_mock_drivers/test/resource/hello_test_bag.bag", true);
```

Comment:
See if we can set this as a relative path.

Reply:
Bag file is included in PR, so it should be possible to setup the relative path.